### ISSUE TITLE
glibc update problem

### ISSUE THREAD

**Issue by acowley** [NixOS/nixpkgs#13629]
The `nixpkgs-unstable` channel hasn't updated in more than two weeks; it is currently blocked by an issue in the Linux boot strap tools. The [log](http://hydra.nixos.org/build/32398519/nixlog/3) suggests it is a problem with a new glibc not being used properly.

**Comment by domenkozar** [comment]
cc @edolstra @vcunat it's the new glibc failing to bootstrap

**Comment by vcunat** [comment]
It's just failing to create (correct new) bootstrap tools. We could make the channel not depend on that in the meantime.

**Comment by vcunat** [comment]
Aha, localized the problem:
```
$ cat ./result/pack/lib/libm.so
/* GNU ld script
*/
OUTPUT_FORMAT(elf64-x86-64)
GROUP ( /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-glibc-2.23/lib/libm.so.6  AS_NEEDED ( /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-glibc-2.23/lib/libmvec_nonshared.a /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-glibc-2.23/lib/libmvec.so.1 ) )
```

**Comment by vcunat** [comment]
However on 16.03 there's a different problem before getting to this one:
```
Unpacking the bootstrap tools...
Patching the bootstrap tools...
patching /nix/store/ab3xql1f1vskhg9z1a1a4y0g45ngvqn0-bootstrap-tools/bin/[
...
patching /nix/store/ab3xql1f1vskhg9z1a1a4y0g45ngvqn0-bootstrap-tools/bin/patchelf
Bus error
```

I've got no idea what that's about.

**Comment by vcunat** [comment]
@edolstra: I'm afraid the new patchelf is broken in some way. When I cherry-pick 0.8 -> 0.9 to master, it also starts failing with that bus error.

**Comment by edolstra** [comment]
Thanks, I'll check it out.

**Comment by edolstra** [comment]
Fixed in a6d19c28f158722ce650a3a1ede352ee4179cdcf.
